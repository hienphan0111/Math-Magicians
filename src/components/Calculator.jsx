import React from 'react';
import './Calculator.scss';

const Calculator = () => (
  <div className="cal-container">
    <span id="result">0</span>
    <span id="ac">AC</span>
    <span id="plus-minus">&#177;</span>
    <span id="mod">%</span>
    <span id="divide">&#247;</span>
    <span id="multi">x</span>
    <span id="minus">-</span>
    <span id="plus">+</span>
    <span id="equal">=</span>
    <span id="9">9</span>
    <span id="8">8</span>
    <span id="7">7</span>
    <span id="6">6</span>
    <span id="5">5</span>
    <span id="4">4</span>
    <span id="3">3</span>
    <span id="2">2</span>
    <span id="1">1</span>
    <span id="zero">0</span>
    <span id="dot">.</span>
  </div>
);

export default Calculator;
