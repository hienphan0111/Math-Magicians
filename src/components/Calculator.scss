.cal-container {
  width: 400px;
  height: 500px;
  margin: 10px auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(6, 1fr);
  justify-content: center;
  align-items: center;
  border: 1px solid #000;
  cursor: pointer;
  font-size: 18px;
  font-weight: 500;

  #result {
    grid-column-start: 1;
    grid-column-end: span 4;
    background-color: rgba(128, 128, 128, 0.567);
    height: 100%;
    display: flex;
    justify-content: flex-end;
    padding-right: 10px;
    font-size: 20px;
  }

  #multi {
    grid-column-start: 4;
    grid-row-start: 3;
  }

  #plus {
    grid-column-start: 4;
    grid-row-start: 4;
  }

  #minus {
    grid-column-start: 4;
    grid-row-start: 5;
  }

  #equal {
    grid-column-start: 4;
    grid-row-start: 6;
  }

  #zero {
    grid-column-start: 1;
    grid-row-start: 6;
    grid-column-end: span 2;
  }

  #one {
    grid-column-start: 1;
    grid-row-start: 5;
  }

  #two {
    grid-column-start: 2;
    grid-row-start: 5;
  }

  #four {
    grid-column-start: 1;
    grid-row-start: 4;
  }

  
  #five {
    grid-column-start: 2;
    grid-row-start: 4;
  }

  #seven {
    grid-column-start: 1;
    grid-row-start: 3;
  }

  #eight {
    grid-column-start: 2;
    grid-row-start: 3;
  }

  .orange {
    background-color: orange;
    border: 0.7px solid #000;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  }

}

.border-center {
  border: 0.5px solid #000;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}
