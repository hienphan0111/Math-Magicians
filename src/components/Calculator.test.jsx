import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import Calculator from './Calculator';
import * as cal from 'logic/calculate';

describe('Calculator component', () => {
  it('Matches DOM Snapshot', () => {
    const component = renderer.create(<Calculator />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render AC button', () => {
    render(<Calculator />);
    const element = screen.getByRole('button', { name: /ac/i });
    expect(element).toBeInTheDocument();
  });

  it('should render call calculate method and display result on screen', () => {
    const spyCalculator = jest.spyOn(cal, "default").mockImplementation(() => ({ total: null, next: 22, operation: null}));
    render(<Calculator />);

    const buttonElement = screen.getByRole('button', {name: /2/i });
    userEvent.click(buttonElement);
    userEvent.click(buttonElement);
    expect(spyCalculator).toBeCalledTimes(2);
    const element = screen.getByText('22');
    expect(element).toBeInTheDocument();
  });
});

