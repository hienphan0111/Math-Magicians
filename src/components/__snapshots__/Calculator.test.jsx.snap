// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Matches DOM Snapshot 1`] = `
<div
  className="cal-container"
>
  <span
    className="border-center"
    id="result"
  >
    0
  </span>
  <span
    className="border-center"
    id="ac"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    AC
  </span>
  <span
    className="border-center"
    id="plus-minus"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    +/-
  </span>
  <span
    className="border-center"
    id="mod"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    %
  </span>
  <span
    className="border-center"
    id="divide"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    รท
  </span>
  <span
    className="border-center"
    id="multi"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    x
  </span>
  <span
    className="border-center"
    id="plus"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    +
  </span>
  <span
    className="border-center"
    id="minus"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    -
  </span>
  <span
    className="border-center"
    id="equal"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    =
  </span>
  <span
    className="border-center"
    id="dot"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    .
  </span>
  <span
    className="border-center"
    id="zero"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    0
  </span>
  <span
    className="border-center"
    id="one"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    1
  </span>
  <span
    className="border-center"
    id="two"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    2
  </span>
  <span
    className="border-center"
    id="three"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    3
  </span>
  <span
    className="border-center"
    id="four"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    4
  </span>
  <span
    className="border-center"
    id="five"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    5
  </span>
  <span
    className="border-center"
    id="six"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    6
  </span>
  <span
    className="border-center"
    id="seven"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    7
  </span>
  <span
    className="border-center"
    id="eight"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    8
  </span>
  <span
    className="border-center"
    id="nine"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    9
  </span>
</div>
`;
